### PR TITLE
Offer 'generate constructor' when a derived type doesn't include a delegating constructor to it's base type

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes: https://github.com/dotnet/Roslyn/issues/6541

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateConstructor/GenerateConstructorTests.cs [hunk 1]
```diff
@@ -693,5 +693,69 @@ public Base(bool val = false)
...
+    public Base(int a, string value = null)
+    {
+    }
+} 
+
+class Derived : Base
+{
+    public Derived(int a, string value = null) : base(a, value)
+    {
+    }
```
Comment (rchande) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateConstructor/GenerateConstructorTests.cs]: Maybe a test for the internal case?

**Comment by rchande** [comment]
:+1:

**Comment by jmarolf** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
Fixes: #6541

File: src/Features/CSharp/Portable/GenerateMember/GenerateConstructor/CSharpGenerateConstructorService.cs [hunk 2]
```diff
@@ -54,6 +60,35 @@ protected override bool IsConstructorInitializerGeneration(SemanticDocument docu
...
+            token = default(SyntaxToken);
+            typeToGenerateIn = null;
+            delegatedConstructor = null;
+
+            var semanticModel = document.SemanticModel;
+            var classDeclaration = (ClassDeclarationSyntax)node;
+            var classSymbol = semanticModel.GetDeclaredSymbol(classDeclaration, cancellationToken);
+
+            var baseType = classSymbol.BaseType;
+            var constructor = baseType.Constructors.FirstOrDefault(c => IsSymbolAccessible(c, document));
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/GenerateMember/GenerateConstructor/CSharpGenerateConstructorService.cs]: Do we want to order based on number of parameters or something like that?

File: src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs [hunk 3]
```diff
@@ -41,22 +41,28 @@ private partial class Editor
 
             internal async Task<Document> GetEditAsync()
             {
-                // First, see if there's an accessible base constructor that would accept these
+                // First, if we were just given the constructor and the type to generate it into
+                // then just go generat that constructor.  There's nothing special we need to do.
```
Comment (Pilchie) [inline src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs]: Spelling.

File: src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs [hunk 4]
```diff
@@ -88,6 +94,51 @@ private partial class Editor
...
+                    _state.TypeToGenerateIn,
+                    new List<ISymbol> { constructor },
+                    new CodeGenerationOptions(_state.Token.GetLocation()),
+                    _cancellationToken)
+                    .ConfigureAwait(false);
+
+                return result;
+            }
+
+
```
Comment (Pilchie) [inline src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs]: Two blank lines.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**